packages/apps-config/src/endpoints/productionRelayKusama.ts

```diff
@@ -20,10 +21,11 @@ export function createKusama (t: TFunction): EndpointOption {
     info: 'kusama',
     text: t('rpc.kusama.parity', 'Kusama', { ns: 'apps-config' }),
     providers: {
-      Parity: 'wss://kusama-rpc.polkadot.io',
-      OnFinality: 'wss://kusama.api.onfinality.io/public-ws',
-      'Patract Elara': 'wss://kusama.elara.patract.io',
-      Pinknode: 'wss://rpc.pinknode.io/kusama/explorer'
+      Parity: createProviderUrl('wss://kusama-rpc.polkadot.io'),
+      OnFinality: createProviderUrl('wss://kusama.api.onfinality.io/public-ws'),
+      'Patract Elara': createProviderUrl('wss://kusama.elara.patract.io'),
+      Pinknode: createProviderUrl('wss://rpc.pinknode.io/kusama/explorer'),
+      'light client': createProviderUrl('kusama-substrate-connect', 'substrate-connect')
```

Comment:
Instead of using `createProviderUrl` why not simply push through only `kusama-substrate-connect` (ends with the `-substrate-connect` as used elsewhere) or something like `substrate-connect://kusama-substrate-connect` and then handling it in actual expansion? 

The light client/ws tests below then just filters on `startWith('substrate-connect')` or `endsWith('-substrate-connect')` depending on what is used.

Reply:
That would be a solution but isn't it extremely "hack-y"? Adding the "type" as a parameter, allows for different kinds to be integrated in the future - meaning: existing (and default): `type: 'json-rpc'`, light-client of substrate: `type: 'substrate-connect'`, some other awesome light client that does not exist yet: `type: 'awesome-light'`. It is better I think than string parsing. In addition light client/ws tests are already fixed based on the string (almost) as you propose

Reply:
We had a long debate about this and thought that overloading the idea of a URL with a custom non-standard scheme was a hack since it would be surprising for people unfamiliar with the code and didn't really accurately reflect what was going on.  In the case where you connect with a light client you are not really using a URL at all - it's conceptually completely different.  Furthermore we are going to need to extend how substrate-connect is configured to support parachains (coming soon!) If we want to add support for that to polkadotjs apps too then there will no sensible way to capture that extra configuration in a URL, but it could neatly be added to the new `Endpoint` type.

Reply:
I hear you, but making calls here is not the way to go. If you want a specific extension, then add it instead of wrapping everything. (This already conflicts with PRs that have been subsequently opened as it stands and just gets worse since it has a lot of churn)

Even as it is specified, it is utterly confusing. So I you want something specific, then rather be explicit on the singular items instead of wrapping all the ones that already do work. So you can make it a `string | SomeType` and just for the handful, specific `SomeType`.

Reply:
Yeah using a union type makes total sense. You get the advantage of not messing with existing URLs and the advantage of a richer type for substrate connect

Reply:
> So you can make it a `string | SomeType` and just for the handful, specify `SomeType` to whatever it may be.

I also agree with that. I will proceed with all needed changes for that